src/session/session_api.c

```diff
@@ -1475,6 +1476,26 @@ __wt_session_range_truncate(
               session, EINVAL, "the start cursor position is after the stop cursor position");
     }
 
+    /*
+     * Utilize the cursor lower bound as the temporary buffer to storing the original start and stop
```

Comment:
A few considerations for this. I don't mind the overloading. I might suggest that it changes to a union then and you use some defines (so that other code doesn't need to change). See `src/btree/bt_salvage.c:90` for an example of what I'm suggesting. I think it would be easier to see `start->original_key` rather than `start->lower_bound`.

Reply:
sorry @sueloverso oops! I wanted to get @luke-pearson 's thoughts on this. Personally I see both benefits of it being easier to read, but at the same time adding a union does the complicate the API in terms of understanding what a lower_bound and upper_bound is. @luke-pearson what are your thoughts on this?

Reply:
If I understand it correctly the suggestion is to create a union of the format:
```
union {
    WT_ITEM *original_key;
    WT_ITEM *lower_bound;
} u
```
I can see advantages to both but I feel like we're adding complexity to the bounded cursor implementation from a change that isn't directly related to it and I'm not a huge fan of that.

I'd rather keep the cursor bound implementation simple even if it means having an oddly named variable in the truncate logging implementation.

Reply:
That understand is correct but incomplete. The other part would be to also add (again similar to `bt_slvg.c`):
#undef lower_bound
#define lower_bound u.lower_bound
#under original_key
#define original_key u.original_key

Doing that should then require no changes at all to the bounded cursor implementation.

Reply:
Thinking about this more, if using that field is the right path then the field should be appropriately named for all uses and any changes needed for all usage should be done. I.e. maybe call it `special_item` or something. Having `lower_bound` in this unrelated code is confusing and will not be remembered years from now. Or we can waste some space and just add another field for `orig_key`. Or a union where mutually exclusive operations can use the same space in different ways.

I don't think that making edits to cursor bound code makes it more complex.

Reply:
If there are no functional changes with the union + hash define implementation I'm okay with that solution.